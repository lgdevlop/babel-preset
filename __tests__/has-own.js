import options from '../babel.config'
import { transformAsync } from '@babel/core'

describe('Object.hasOwn test suite', () => {
    test('Should add polyfills import for Object.hasOwn built-in usage', async () => {
        const input = `Object.hasOwn({abc: 'abc'}, 'abc')`

        const { code: output } = await transformAsync(input, {
            ...options,
            filename: 'file.ts'
        })
        expect(output).toMatchSnapshot()

        const evaluated = eval(output)

        expect(evaluated).toEqual(true)
    })
})
