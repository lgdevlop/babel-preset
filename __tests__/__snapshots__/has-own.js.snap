// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Object.hasOwn test suite Should add polyfills import for Object.hasOwn built-in usage 1`] = `
"\\"use strict\\";

require(\\"core-js/modules/es.object.has-own.js\\");

Object.hasOwn({
  abc: 'abc'
}, 'abc');"
`;
